### PR TITLE
Add helpers and instructions on how to share datasets on the huggingface hub

### PR DESCRIPTION
**Pull request by ernestum**
## Description

I added some helpers to make it easier to publish or download datasets with trajectories from HuggingFace Datasets Hub.

## Testing

Not tested so far! Add coverage for:
- [x] getting a slice out of a HuggingFace Datasets backed trajectories sequence
- [ ] The trivial case where we convert a sequence of trajectories to a HuggingFace Dataset, that is already backed by a HuggingFaceDataset

### CODE REVIEW

File: docs/tutorials/trajectories.rst [hunk 1]
```diff
@@ -65,3 +65,33 @@ In the same way you can load trajectories from a HuggingFace Dataset:
...
+.. code-block:: python
+
+    import datasets
+    from imitation.data.huggingface_utils import TrajectoryDatasetSequence
+
+    your_dataset = datasets.load_dataset("your_hf_name/your_dataset_name")
+    your_trajectories = TrajectoryDatasetSequence(your_dataset)
+
+The :class:`TrajectoryDatasetSequence <imitation.data.huggingface_utils.TrajectoryDatasetSequence>`
+wraps a HuggingFace dataset so it can be used in the same way as a list of trajectories.
```
Comment (timbauman) [inline docs/tutorials/trajectories.rst:97]: Might be nice to include an example of passing it into a method that takes a sequence of trajectories

File: src/imitation/data/huggingface_utils.py [hunk 2]
```diff
@@ -174,3 +179,14 @@ def trajectories_to_dict(
             cast(types.TrajectoryWithRew, traj).rews for traj in trajectories
         ]
     return trajectory_dict
+
+
+def trajectories_to_dataset(
```
Comment (timbauman) [inline src/imitation/data/huggingface_utils.py:184]: 👍

**Review by timbauman — COMMENTED** [review]
Hey! Tim again, reviewing this PR 😄

I'm sure you're aware but you can make this a PR based on https://github.com/HumanCompatibleAI/imitation/pull/704 by changing the base branch from `master` to `huggingface_datasets_documentation` - then it'll only include changes from this PR and not that one

File: docs/experts/loading-experts.rst [hunk 3]
```diff
@@ -26,11 +26,12 @@ corresponding policy loader.
...
 .. code-block:: python
 
+    import numpy as np
     from imitation.policies.serialize import load_policy
     from imitation.util import util
 
-    venv = util.make_vec_env("your-env", n_envs=4)
-    local_policy = load_policy("ppo", venv, loader_kwargs={"path": "path/to/model.zip"})
+    venv = util.make_vec_env("your-env", n_envs=4, rng=np.random.default_rng())
+    local_policy = load_policy("ppo", venv, path="path/to/model.zip")
```
[the remark this answers is not in the document]
Comment (ernestum) [reply docs/experts/loading-experts.rst:34]: "always" is a big word here :-) but this PR did not introduce it.

I just noticed that in this example, an old version of the `load_policy` API was used where the kwargs to be passed to the loaded still needed to be wrapped inside a `loader_kwargs` dict and fixed it here.

File: src/imitation/testing/hypothesis_strategies.py [hunk 4]
```diff
@@ -0,0 +1,118 @@
+"""Hypothesis strategies for generating sequences of trajectories for testing."""
```
[the remark this answers is not in the document]
Comment (ernestum) [reply src/imitation/testing/hypothesis_strategies.py:1]: Yes. I already had this in the back of my mind but never spelled it out somewhere. Now you will find it in #717.

**Pull request merged**